admin_guide/cluster_management.rst

```diff
@@ -82,3 +82,64 @@ place for the master node, users can dynamically reconstruct this metadata from
 shard information available on the worker nodes. To learn more about the metadata
 tables and their schema, you can visit the :ref:`metadata_tables` section of our documentation.
 
+Tenant Isolation
+################
+
+.. note::
+
+  Tenant isolation is a feature of **Citus Enterprise Edition** and :ref:`Citus Cloud <cloud>` only.
+
+Citus places table rows into worker shards based on the hashed value of the rows' distribution column. Multiple distribution column values often fall into the same shard. In the Citus multi-tenant use case this means that tenants often share shards.
+
+Many multi-tenant SaaS providers face the problem that some tenants grow much bigger than the others, or require special resource guarantees. In Citus Enterprise Edition large tenants can be *isolated* to their own dedicated shards to better control the resource allocation between small and large tenants.
```

Comment:
Could we add more context on the motivation. Maybe say something like:- "When you have a large number of tenants, not all tenants are equally sized. Generally, we've seen that data follows a Zipf's distribution where you would have a few tenants which have a large amount of data and a large number of smaller sized tenants. Once your bigger tenants grow to a large size, it might be worthwhile to move them to a different node. This would allow you to provide better resource allocation and guarantees to larger tenants and also give more resources to other smaller tenants.  To enable this, Citus enterprise edition allows you to isolate ......